client/src/main/scala/com/all4journey/webapp/util/Fragments.scala

```diff
@@ -0,0 +1,151 @@
+        div(id := "zipCodeHelpBlock"),
+        input(id := "zipCode", name := "zipCode", cls := "form-control partOfZipCodeList", `type` := "text", value := address.getOrElse(emptyAddress).zipCode)
+      )
+    )
+  )
+
+  def setUiViolationPrompts(violations: Set[Violation]) = {
+    if (!violations.isEmpty) {
+      //                      val content = dom.document.getElementById("content")
+      //                      for (v <- failureList)
+      //                        content.appendChild(p(v.description).render)
+      violations.foreach(violation =>
+        if (violation.description.getOrElse("").contains("street")) {
+          $("#streetAddressDiv").addClass("has-error")
+          val streetAddressHelpBlock = dom.document.getElementById("streetAddressHelpBlock")
+          streetAddressHelpBlock.appendChild(span(cls := "help-block")("invalid street address format").render)
+        }
+        else if (violation.description.getOrElse("").contains("state")) {
+          HtmlHelper.showHelpBlock("#stateDiv", "stateHelpBlock", "please choose a state")
+        }
+        else if (violation.description.getOrElse("").contains("zipCode")) {
+          HtmlHelper.showHelpBlock("#zipCodeDiv", "zipCodeHelpBlock", "invalid zip code format")
+
+        }
+      )
+    }
+  }
+
+  def validateAddress(address: Address): Set[Violation] = {
+    var violations = Set.empty[Violation]
+    val isFullAddressValid = validate(address)(Address.validatorWithStreet) match {
+      case ValidationSuccess   => true
+      case Failure(failureSet) =>
+        violations = failureSet
+        false
+    }
+
+    if (!isFullAddressValid) {
+      val partialAddressValidationResult = if (address.street.getOrElse("").isEmpty && !address.state.id.equals("NONE"))
+        validate(address)(Address.validatorNoStreetWithState)
+      else if (address.street.getOrElse("").isEmpty && address.state.id.equals("NONE") && (address.zipCode != null && !address.zipCode.isEmpty))
+        validate(address)(Address.validatorNoStreetNoStateWithZip)
+      else if (address.street.getOrElse("").isEmpty && address.state.id.equals("NONE") && (address.zipCode == null || address.zipCode.isEmpty))
+      // an empty address is a valid address
+        ValidationSuccess
+
+      violations = partialAddressValidationResult match {
+        case ValidationSuccess   => Set.empty[Violation]
+        case Failure(failureSet) => failureSet
+        case _ => violations
+      }
+    }
+
+    violations
+  }
+
+  def buildAddressFromForm(): Address = {
+    val addressUuid = $("#addressFieldsDiv").attr("addressUuid").toString.trim
+    val streetAddress = $("#streetAddress").value().toString.trim
+    val stateId = $("#userState").value().toString.trim
+    val zipCode = $("#zipCode").value().toString.trim
+
+    new Address(addressUuid, "0", Option(streetAddress), State(stateId, ""), zipCode, HomeAddressType, "Home")
+  }
+
+  def refreshAddressForm(data: js.Any): Unit = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [You should not name methods after their defining object](https://www.codacy.com/app/rjkernick/all4journey/file/2503283144/issues/source?bid=3246460&fileBranchId=3273764#l125)